### PR TITLE
MIPI Display, GNU Linker, Bug Fixes

### PR DESCRIPTION
**Pull request by arashed-alif**
Bug Fixes: use static keyword on mpu_table and __mram_atoc to prevent linker from removing sections
MIPI Display: changes to stubs, delay function, cdc200 driver, dsi host driver
GNU Linker: re-arrange buffers to share SRAM with KWS demo on RTSS-HE, properly set .copy.table and .zero.table

### CODE REVIEW

**Comment by arashed-alif** [comment]
Leave any comments or give it the OK to merge.

**Comment by arashed-alif** [comment]
Please see the commit messages for explanation of the changes. The latest change to the linker is notable.

File: scripts/cmake/toolchains/bare-metal-armclang.cmake [hunk 1]
```diff
@@ -59,6 +59,9 @@ add_compile_options(
 # General purpose compile options:
 add_compile_options(
     -funsigned-char
+    -Og
```
Comment (kivaisan) [inline scripts/cmake/toolchains/bare-metal-armclang.cmake:62]: Are these needed anymore? Should normal cmake debug build be enough?

**Review by jarvte — APPROVED** [review]
Did not check the display changes, other changes look ok. So this works now that nn_model is in mram, that's cool!

File: scripts/cmake/toolchains/bare-metal-armclang.cmake (hunk 1 above)
Comment (arashed-alif) [reply scripts/cmake/toolchains/bare-metal-armclang.cmake:62]: I'll send a commit to remove it.

**Pull request merged**